.github/workflows/linux-workflow.yml

```diff
@@ -74,7 +74,7 @@ jobs:
 
       - name: Checkout Submodules
         if: steps.cache-submodules.outputs.cache-hit != 'true'
-        run: git submodule update --init --recursive --jobs 2
+        run: git submodule update --init --recursive --depth 1 -j $(nproc)
```

Comment:
why? according to git documentation depth is used to create a shallow clone of only the latest commit, with in this case only the master branch. As git submodules can, in theory, target branches, then that's not something applicable everywhere. nproc doesn't do much here either as it's not a CPU intensive task, but a network bound one, mostly.

Reply:
In that case shouldn't we use https://git-scm.com/docs/git-submodule#Documentation/git-submodule.txt---no-recommend-shallow instead?

Reply:
Yes I know they aren't since we currently do a full clone, my point was wouldn't that be the best option there, in case we do need a non shallow clone?

Reply:
Sure, You can go ahead and do it in this PR? thanks!

Reply:
I pushed the changes to .gitmodules.